stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -459,7 +459,13 @@ extension ${Self}: BinaryFloatingPoint {
     precondition(payload < ${Self}._quietNaNMask,
                  "NaN payload is not encodable.")
     var significand = payload
-    if !signaling { significand |= ${Self}._quietNaNMask }
+    if signaling {
+      precondition(payload != 0,
```

Comment:
We can't use `precondition()` in the library, it won't work correctly.  Please use `_precondition()`.